main.go

```diff
@@ -203,10 +235,24 @@ func main() {
 			// If leader election is not enabled, we can always return ready
 			return healthz.Ping(req)
 		}
+
+		// Leader is always ready when initialized
 		if isLeader.Load() && isCloudManagerInitialized.Load() {
-			// If the manager is the leader, we can return nil to indicate readiness
 			return nil
 		}
+
+		// CRITICAL FIX: Allow healthy non-leader pods to report ready as standbys.
```

Comment:
Need to analyze the impact of this change.

Reply:
Reverted this change to avoid duplicated webhook eps and cache inconsistance.
Deadlock during deployment patch can be avoided by setting `maxUnavailable` to 100%